From stable-bounces@linux.kernel.org  Fri Feb  3 20:04:58 2006
Date: Fri, 3 Feb 2006 19:57:49 -0800
From: Andrew Morton <akpm@osdl.org>
To: stable@kernel.org
Cc: 
Subject: [PATCH] SELinux: fix size-128 slab leak

From: Stephen Smalley <sds@epoch.ncsc.mil>

Remove private inode tests from security_inode_alloc and security_inode_free,
as we otherwise end up leaking inode security structures for private inodes.

Signed-off-by:  Stephen Smalley <sds@tycho.nsa.gov>
Acked-by: James Morris <jmorris@namei.org>
Signed-off-by:  Linus Torvalds <torvalds@osdl.org>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 include/linux/security.h |    4 ----
 1 files changed, 4 deletions(-)

Index: linux-2.6.15.3/include/linux/security.h
===================================================================
--- linux-2.6.15.3.orig/include/linux/security.h
+++ linux-2.6.15.3/include/linux/security.h
@@ -1437,15 +1437,11 @@ static inline void security_sb_post_pivo
 
 static inline int security_inode_alloc (struct inode *inode)
 {
-	if (unlikely (IS_PRIVATE (inode)))
-		return 0;
 	return security_ops->inode_alloc_security (inode);
 }
 
 static inline void security_inode_free (struct inode *inode)
 {
-	if (unlikely (IS_PRIVATE (inode)))
-		return;
 	security_ops->inode_free_security (inode);
 }
 
